streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -738,7 +737,14 @@ private Queue<ProducerRecord<byte[], byte[]>> getRecordsQueue(final String topic
     public final <K, V> TestInputTopic<K, V> createInputTopic(final String topicName,
```

Comment:
Also wondering why all these methods are final -- I could have addressed this issue in our code without this PR if I could just override these methods. Are we concerned about users doing this..? And why?

I figure this change actually would need a KIP so I left it out, but I'd like to remove the `final` from these unless someone knows a good reason for doing this

Reply:
You are right, you'd need a KIP for removing it. I'm fine with merging this fix without removing the `final`. 

For removing the `final` -- it does make sense to restrict the surface area for extending the public interface. I'm not sure why we'd need to override those. But this is something for the KIP discussion anyways, if you want to write one